vulkaninfo/vulkaninfo.c

```diff
@@ -4572,16 +4558,11 @@ static void AppGpuDumpQueueProps(const struct AppGpu *gpu, uint32_t id, FILE *ou
         printf("\t\t\t\"queueCount\": %u,\n", props.queueCount);
         printf("\t\t\t\"queueFlags\": %u,\n", props.queueFlags);
         printf("\t\t\t\"timestampValidBits\": %u,\n", props.timestampValidBits);
-        printf("\t\t\t\"present_support\": {\n");
-        for (struct SurfaceNameWHandle *surface_stuff = gpu->inst->surface_chain; surface_stuff != NULL;
-             surface_stuff = surface_stuff->pNextSurface) {
-            if (surface_stuff->pNextSurface) {
-                printf("\t\t\t\t\"%s\" : %u,\n", surface_stuff->name, surface_stuff->present_support);
-            } else {
-                printf("\t\t\t\t\"%s\" : %u\n", surface_stuff->name, surface_stuff->present_support);
-            }
+        printf("\t\t\t\"present_support\": [\n");
+        for (struct SurfaceExtensionNode *sen = gpu->inst->surface_ext_infos_root; sen != NULL; sen = sen->next) {
+            printf("\t\t\t\t\"%s\": \"%s\"%s\n", sen->name, sen->supports_present ? "true" : "false", sen->next ? "," : "");
```

Comment:
@mikew-lunarg can correct me on this, but shouldn't this to be a 1 or 0 instead of a "true" or "false" string since it is JSON output?

Reply:
json has bool this way. Not sure about conventions in this repo about it...

Reply:
@jeremyk-lunarg @krOoze  vulkaninfo's JSON was specifically designed to be consumed by the device simulation (devsim) layer.   The [devsim JSON schema](https://schema.khronos.org/vulkan/devsim_1_0_0.json) defines its types directly from vulkan_core.h so Vulkan's booleans, which are VkBool32 and derived from uint32_t, should be integer values.

If there isn't already, there should be a comment in vulkaninfo that the generated JSON must validate against the devsim schema at https://schema.khronos.org/vulkan/devsim_1_0_0.json

Reply:
That's fine with me. Features seem to be binary numbers too...